langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolExecutionRequestUtil.java

```diff
@@ -14,7 +16,7 @@ private ToolExecutionRequestUtil() {}
     /**
      * Gson instance.
      */
-    public static final Gson GSON = new Gson();
+    public static final Gson GSON = new GsonBuilder().setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE).create();
```

Comment:
I would be very careful with this change, it might affect a lot. Is there a way to avoid this change and/or do it only for Ollama?

Reply:
OK i will revert it, but without it, integer or long are transformed into double and you need to convert it back to int following parameters.

Reply:
I am just not sure about the impact on all other modules. If you believe this change is necessary, let's create another issue and make a change with thorough analysis that there are no major impact on all modules who use this code, WDYT?

Reply:
ok , will do a specific PR for this change. 
In any case i've changed the different Unit test impacted by this change, but not sure for the IT.
Is there a procedure to trigger the full integration tests. I don't have keys for the specific providers.

Reply:
@humcqc there is no way to run ITs right now in PRs, this is being addressed in https://github.com/langchain4j/langchain4j/pull/1185

You can run all ITs locally by executing `mvn verify`
Most (if not all) ITs are guarged by `@EnabledIfEnvironmentVariable(named = "*_API_KEY", matches = ".+")`
So if there is no key, they will be skipped

Reply:
[Here](https://github.com/langchain4j/langchain4j/pull/1394) is a dedicated PR for this